pkg/project/namespace.go

```diff
@@ -93,6 +94,15 @@ func (ns *Namespaced) Validate() error {
 	return nil
 }
 
+func (ns *Namespaced) Equal(other *Namespaced) bool {
+	if ns == nil || other == nil {
+		return false
+	}
+
+	return !ns.AllowOmitOwner && strings.EqualFold(strings.ToLower(other.String()), strings.ToLower(ns.String())) ||
```

Comment:
This has multiple issues:

* You're not checking whether AllowOmitOwner matches. And `Equal()` makes no implications about this behavior.
* Either namespace can include commit ID's. The current behavior would want the commit ID's  to match also. 

I suggest naming this something a bit more explicit. eg. `OwnerIsEqual()` and `ProjectIsEqual()`.

Reply:
Rather than having a convenience function I've moved the checks back to where they were initially.